swirl_lm/boundary_condition/immersed_boundary_method.py

```diff
@@ -1124,6 +1699,252 @@ def _apply_feedback_force_1d_interp(
 
     return ib_force
 
+  def _ib_gp_force_fn(
+      self,
+      interp_weights: FlowFieldVal,
```

Comment:
The selection of these weights depends on the type of BC we'd like to enforce. One of them is redundant regardless of the type of boundary conditions. Let's make it an argument of the returned function to eliminate the ambiguity.